packages/@expo/cli/src/utils/mergeGitIgnorePaths.ts

```diff
@@ -131,7 +130,35 @@ export function upsertGitIgnoreContents(
   if (targetGitIgnore.includes(generatedHeaderPrefix)) {
     const indexes = getGeneratedSectionIndexes(targetGitIgnore);
 
-    contents = `${indexes.contents.slice(indexes.start + 3, indexes.end).join('\n')}\n${contents}`;
+    if (indexes.start > -1 && indexes.end > -1 && indexes.start < indexes.end) {
```

Comment:
The merge functionality here was not working as expected, this is for updating the following section that is automatically added to gitignore.

<img width="714" height="159" alt="Screenshot 2025-08-28 at 23 36 47" src="https://github.com/user-attachments/assets/6b82893c-1372-4ee7-8fe4-70c965a05000" />

Previously if I removed `/ios` and ran prebuild, I would get prompted to add `/android`, but it would not get added. This ensures that both previous and current values are merged correctly.